### DISCUSSION TITLE
No Fail Alterations (Score Modifier Only on Fail)

### DISCUSSION THREAD

**Discussion by Kevin2Holt** [ppy/osu#28706] [Ideas] [answered]
As a super casual player, I much prefer to just play through an entire song every time rather than failing and starting over halfway through. So, I play with No Fail (NF) on pretty much all the time. However, this also puts a .5x multiplier on my scores which basically completely takes me out of any sort of ranking.

### My adjustment to the way NF works:
**Only apply the .5x multiplier **if/when you actually fail**, not as a blanket _"if-you-have-the-mod-on"_.**

Having no understanding of how the back-end actually works (I've never looked into it, though I am a programmer), I feel like this might require a slight modification to the scoring/multiplier system.

There are two ways to approach this:
1. Apply the multiplier to the entire score
2. The multiplier only applies to the part _after_ you fail

My personal preference would be method 2, only applying the multiplier to part of the beatmap (the part before failing giving you the full score). However, I also understand that this would require another _"slight"_ change to the scoring system and potentially mess with some rankings and other connected systems. So, I think that method 1 (applying the multiplier to the entire score, as it does currently, only "activating" the multiplier _after_ failing), would be more straightforward to implement, not affecting the current scores/rankings. _(Yes, this is assuming not changing/updating previous NF plays with a new score)_

**Comment by Stoppedpuma** [comment]
This kind of ruins the point of having a multiplier let alone the mod as everyone will just keep it on at that point.

**Reply by Kevin2Holt** [reply]
1) There is still a multiplier. The difference is when it gets applied
2) The culture of the community is already off of NF. The majority of people will likely use the default (without NF).

I still feel like it's an appropriate request/suggestion. The difference between having it on and not is whether or not you guarantee you're going to play the entire map. I personally prefer to guarantee play through the entire song. That's just more enjoyable to me. I can understand the other too. I used to be there (then I got more casual)...

The point is, if you don't fail, you don't get the score reduction. If you do fail, then your score gets halved. Simple and easy.

Despite being super casual, I just want to have the opportunity to be properly ranked while still playing in the way I enjoy the most (all the way through each song).

**Reply by Stoppedpuma** [reply]
Another issue here is implementation, there's no other mods that work like this and I think it would be odd to just have a single mod determine in the middle of a map if your multiplier is halved of not. This could be pretty confusing to new players, especially on the original point of why not just always keep it on at that point since it guarantees a pass which I feel takes a lot away from difficulty aspects.

**Comment by Tenexxt** [comment]
Imo this makes sense, because if you don't fail, the mod isn't really doing anything lol. This applies to many situations really (like using spun out in a map without spinners), so I doubt they will implement it, Because then everyone would go "If this mod has it, why doesn't this one?!" And the programming hell would unleash on the devs. 
If this would need alterations to the score system, its a smaller problem, because pp and score systems are getting reworked, so they could just add since they are already reworking it.

But yea, I doubt they will add it + if you want to play through the song, play an easier diff (unless there aren't any easier ones, then my bad)

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/discussions/21865
https://github.com/ppy/osu/discussions/18424
https://github.com/ppy/osu/discussions/16779
https://github.com/ppy/osu/discussions/14114
https://github.com/ppy/osu/issues/9844
